benchmarks/adjoint/adjointkernel_degree4.prm

```diff
@@ -0,0 +1,141 @@
+set Pressure normalization                = surface
+set Surface pressure                      = 0
+
+# Choose the adjoint solver scheme and set the number of iterations to 1
+# (i.e. don't iterate over the parameters)
+set Nonlinear solver scheme = Stokes adjoint
+set Max nonlinear iterations = 1
+
+
+subsection Material model
+  # additive means after each iteration we add the value stored in the 
+  # composition field to the existing density / multiply the viscosity 
+  # with the new composition field
+  set Model name = additive
+
+  subsection Additive model
+     set Base model additive = simple
+  end
+
+  subsection Simple model
+    set Thermal expansion coefficient = 4e-5
+    set Viscosity                     = 1e21
+    set Thermal viscosity exponent    = 3
+    set Reference temperature         = 1600
+  end
+
+end
+
+# to run sensibly with stokes adjoint the following things stay:
+# Composition polynomial degree
+# Additive model
+# 2 compositional fields with exactly those names 
+subsection Discretization
+  set Composition polynomial degree        = 0
+  set Use discontinuous composition discretization = true
+end
+
+subsection Compositional fields
+  set Number of fields = 2
+  set Names of fields = density_term, viscosity_factor
+end
+
+subsection Initial composition model
+  set Model name = function
+
+  subsection Function
+    # before the iteration the density term should be 0 and the
+    # viscosity prefactor should be 1
+    # this means the addition to density at the first iteration is 0 
+    # while the multiplication to viscosity is 1
+    # these scalar values are prescribed everywhere in 3D field
+    # during iteration the composition value will reflect the incremental addition
+    # multiplication
+    set Function expression = 0 ; 1
+  end
+end
+
+
+
+subsection Geometry model
+  set Model name = spherical shell
+
+  subsection Spherical shell
+    set Inner radius  = 3481000
+    set Outer radius  = 6336000
+  end
+end
+
+
+subsection Boundary velocity model
+    set Zero velocity boundary indicators       =
+  set Tangential velocity boundary indicators = inner, outer
+  set Prescribed velocity boundary indicators =
+end
+
+subsection Boundary temperature model
+  set Fixed temperature boundary indicators   = inner, outer
+end
+
+
+subsection Nullspace removal
+  set Remove nullspace = net rotation
+end
+
+subsection Boundary temperature model
+  set Model name = spherical constant
+  subsection Spherical constant
+    set Inner temperature = 4273
+    set Outer temperature = 973
+  end
+end
+
+
+subsection Initial temperature model
+  set Model name = spherical hexagonal perturbation
+
+  subsection Spherical hexagonal perturbation
+    set Angular mode = 4
+  end
+end
+
+
+subsection Gravity model
+  set Model name = radial constant
+
+  subsection Radial constant
+    set Magnitude = 10
+  end
+
+end
+
+
+subsection Mesh refinement
+  set Initial global refinement          = 5
```

Comment:
Good point - the parameter file is in 2D so it does run quickly. However, the actual benchmark would be in 3D. Should I leave it in 2D and make a note of that or should I change it to 3D and add information on the computational cost?

Reply:
What else should be saved in the benchmark folder? Should it include the output? I can have a look through the other folders.

Reply:
Ah, I missed that that was in 2d. It's fine to leave it as is, and say somewhere in the text of the manual that the .prm runs in 2d, and then show 3d results where you make it clear that one has to change the dimension in the .prm file to reproduce those. That's better than having a super expensive computation in the .prm file because people will try to run it and it will lock up their machine :-)

In the folder, only store what is necessary to run the benchmark, none of the output files other than what you need to show pictures and tables in the manual.